### ISSUE TITLE
0xHati - _whiteListRouters uses token0 and token1 but these are not initialized yet

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-06-arrakis-judging#11]
0xHati

medium

# _whiteListRouters uses token0 and token1 but these are not initialized yet

## Summary
`__whiteListRouters` checks for each router to make sure it's not the address of either token0 or token1. However, the tokens are initialized only after calling `_whiteListRouters`.
## Vulnerability Detail
`_whiteListRouters` has a check to make sure that the provided router is not `token0` nor `token1`, since this will mess up the functionalities of the protocol. However, the check is not doing anything really since it will always pass. The tokens are only set after the function is called so in the function call it will just check if `token0` or `token1` is not `address(0)`
## Code Snippet
[Link to code](https://github.com/sherlock-audit/2023-06-arrakis/blob/main/v2-core/contracts/abstract/ArrakisV2Storage.sol#L134)
[Link to _whiteListRouters](https://github.com/sherlock-audit/2023-06-arrakis/blob/main/v2-core/contracts/abstract/ArrakisV2Storage.sol#L311)
```solidity
 _addPools(params_.feeTiers, params_.token0, params_.token1);
 _whitelistRouters(params_.routers);

 token0 = IERC20(params_.token0);
 token1 = IERC20(params_.token1);
```
```solidity
function _whitelistRouters(address[] calldata routers_) internal {
        for (uint256 i = 0; i < routers_.length; i++) {
            require(
                routers_[i] != address(token0) &&
                    routers_[i] != address(token1),
                "RT"
            );
            require(!_routers.contains(routers_[i]), "CR");
            // explicit.
            _routers.add(routers_[i]);
        }

        emit LogWhitelistRouters(routers_);
    }
  ```
## Tool used

Manual Review

## Recommendation

Duplicate of #179
